server/game/cards/09.1-TCR/HighStakesHaven.js

```diff
@@ -7,19 +7,38 @@ class HighStakesHaven extends DeedCard {
                 onDrawHandsRevealed: () => this.controller.isCheatin()
             },
             handler: () => {
-                if(this.controller.getSpendableGhostRock() <= 0) {
-                    this.havenRandomDiscard();
-                } else if(this.controller.hand.length) {
-                    this.game.promptWithMenu(this.controller, this, {
-                        activePrompt: {
-                            menuTitle: 'Choose cheatin\' punishment',
-                            buttons: [
-                                { text: 'Pay 1 GR', method: 'payGR' },
-                                { text: 'Discard card', method: 'havenRandomDiscard' }
-                            ]
-                        },
-                        source: this
-                    });
+                let choicemask = 0;
```

Comment:
Similar to the `InnerStruggle`, I would rather change `discardAtRandom` function to return if the card was discarded.